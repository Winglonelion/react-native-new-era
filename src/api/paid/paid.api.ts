import axios from 'axios';
import {
  PAID_HISTORY_OVERVIEW_ENDPOINT,
  PAID_OVERVIEW_ENDPOINT,
} from '../api.endpoint';
import {
  GetPaidHistoryOverViewParameters,
  GetPaidHistoryOverViewResponse,
  GetPaidOverViewParameters,
  GetPaidOverViewResponse,
} from './paid.api.types';

export function getPaidOverViewAPI({ userId }: GetPaidOverViewParameters) {
  return axios.post<GetPaidOverViewResponse>(PAID_OVERVIEW_ENDPOINT, {
    user_id: userId,
  });
}

export function getPaidHistoryOverViewAPI({
  userId,
}: GetPaidHistoryOverViewParameters) {
  return axios.post<GetPaidHistoryOverViewResponse>(
    PAID_HISTORY_OVERVIEW_ENDPOINT,
    {
      user_id: userId,
    },
  );
}
