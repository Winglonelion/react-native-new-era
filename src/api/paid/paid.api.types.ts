export type GetPaidOverViewParameters = {
  userId: string;
};

export type GetPaidOverViewResponse = {
  take_home: number;
  gross: number;
  hours: number;
  rate: number;
  regular: number;
  over_time: number;
  detail: { type: string; value: number }[];
};

export type GetPaidHistoryOverViewParameters = {
  userId: string;
};

export type GetPaidHistoryOverViewResponse = {
  date: string;
  take_home: number;
  gross: number;
  hours: number;
}[];
