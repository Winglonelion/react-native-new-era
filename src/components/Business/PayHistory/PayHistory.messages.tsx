const messages = {
  pay_history: 'Pay History',
};

export default messages;
