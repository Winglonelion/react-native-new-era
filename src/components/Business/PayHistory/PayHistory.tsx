import React, { memo } from 'react';
import { Image, StyleSheet, View } from 'react-native';

import Colors from 'theme/colors';
import CommonStyles from 'theme/CommonStyles';
import TitleText from 'components/Base/Text/TitleText';
import { GetPaidHistoryOverViewResponse } from 'api/paid/paid.api.types';
import PayHistoryItem from '../PayHistoryItem';
import messages from './PayHistory.messages';

export interface PropTypes {
  loading?: boolean;
  overviewData?: GetPaidHistoryOverViewResponse;
}

const PayHistoryOverView: React.FC<PropTypes> = memo(
  ({ loading, overviewData: overviewData }) => {
    return (
      <View style={styles.shadowBox}>
        <View style={styles.container}>
          {/* title row */}
          <View style={styles.titleRow}>
            <TitleText level="h2">{messages.pay_history}</TitleText>
            <Image
              source={require('images/chevron_right_corban.png')}
              style={styles.icon}
            />
          </View>
          {loading ? (
            <>
              <PayHistoryItem key="place_holder_1" />
              <PayHistoryItem key="place_holder_2" />
              <PayHistoryItem key="place_holder_3" />
            </>
          ) : (
            overviewData?.map(item => (
              <PayHistoryItem key={item.date} item={item} />
            ))
          )}
          {/* contents */}
        </View>
      </View>
    );
  },
);

const styles = StyleSheet.create({
  shadowBox: {
    width: '100%',
    paddingVertical: 8,
    paddingTop: 2,
    paddingHorizontal: 8,
  },
  container: {
    width: '100%',
    paddingHorizontal: 24,
    paddingRight: 20,
    paddingVertical: 25,
    paddingBottom: 40,
    backgroundColor: Colors.coldLight,
    borderRadius: 8,
    ...CommonStyles.shadow,
  },
  titleRow: {
    height: 40,
    flexDirection: 'row',
    marginBottom: 5,
    justifyContent: 'space-between',
  },
  icon: {
    width: 24,
    height: 24,
    resizeMode: 'contain',
  },
});

export default PayHistoryOverView;
