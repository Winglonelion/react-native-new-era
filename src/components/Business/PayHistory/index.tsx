export { default as default } from './PayHistory';
