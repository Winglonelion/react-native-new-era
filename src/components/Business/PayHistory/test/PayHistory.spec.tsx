import React from 'react';
import { shallow } from 'enzyme';
import PayHistory, { PropTypes } from '../PayHistory';
import { paid_history_overview } from 'mock/networking/paid.mock';

const setup = (propOverrides: PropTypes = {} as PropTypes) => {
  const props = { ...propOverrides };
  const wrapper = shallow(<PayHistory {...props} />);
  return {
    props,
    wrapper,
  };
};

describe('PayHistory check snapshot', () => {
  test('should match snapshot with props', () => {
    const { wrapper } = setup({
      loading: false,
      overviewData: paid_history_overview,
    });
    expect(wrapper).toMatchSnapshot();
  });
  test('should match snapshot without props and loading', () => {
    const { wrapper } = setup({
      loading: true,
    });
    expect(wrapper).toMatchSnapshot();
  });
});
