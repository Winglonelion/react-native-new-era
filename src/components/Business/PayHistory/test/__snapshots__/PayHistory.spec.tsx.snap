// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`PayHistory check snapshot should match snapshot with props 1`] = `
<View
  style={
    Object {
      "paddingHorizontal": 8,
      "paddingTop": 2,
      "paddingVertical": 8,
      "width": "100%",
    }
  }
>
  <View
    style={
      Object {
        "backgroundColor": "#F2F7F7",
        "borderRadius": 8,
        "elevation": 2,
        "paddingBottom": 40,
        "paddingHorizontal": 24,
        "paddingRight": 20,
        "paddingVertical": 25,
        "shadowColor": "rgba(45, 49, 53, 0.18)",
        "shadowOffset": Object {
          "height": 2,
          "width": 0,
        },
        "shadowOpacity": 1,
        "shadowRadius": 2,
        "width": "100%",
      }
    }
  >
    <View
      style={
        Object {
          "flexDirection": "row",
          "height": 40,
          "justifyContent": "space-between",
          "marginBottom": 5,
        }
      }
    >
      <Memo()
        level="h2"
      >
        Pay History
      </Memo()>
      <Image
        source={
          Object {
            "testUri": "../../../src/images/chevron_right_corban.png",
          }
        }
        style={
          Object {
            "height": 24,
            "resizeMode": "contain",
            "width": 24,
          }
        }
      />
    </View>
    <Memo()
      item={
        Object {
          "date": "2021-01-31",
          "gross": 2400,
          "hours": 180,
          "take_home": 2200.01,
        }
      }
      key="2021-01-31"
    />
    <Memo()
      item={
        Object {
          "date": "2021-02-28",
          "gross": 2400,
          "hours": 180,
          "take_home": 2200.01,
        }
      }
      key="2021-02-28"
    />
    <Memo()
      item={
        Object {
          "date": "2021-01-15",
          "gross": 2400,
          "hours": 180,
          "take_home": 2200.01,
        }
      }
      key="2021-01-15"
    />
  </View>
</View>
`;

exports[`PayHistory check snapshot should match snapshot without props and loading 1`] = `
<View
  style={
    Object {
      "paddingHorizontal": 8,
      "paddingTop": 2,
      "paddingVertical": 8,
      "width": "100%",
    }
  }
>
  <View
    style={
      Object {
        "backgroundColor": "#F2F7F7",
        "borderRadius": 8,
        "elevation": 2,
        "paddingBottom": 40,
        "paddingHorizontal": 24,
        "paddingRight": 20,
        "paddingVertical": 25,
        "shadowColor": "rgba(45, 49, 53, 0.18)",
        "shadowOffset": Object {
          "height": 2,
          "width": 0,
        },
        "shadowOpacity": 1,
        "shadowRadius": 2,
        "width": "100%",
      }
    }
  >
    <View
      style={
        Object {
          "flexDirection": "row",
          "height": 40,
          "justifyContent": "space-between",
          "marginBottom": 5,
        }
      }
    >
      <Memo()
        level="h2"
      >
        Pay History
      </Memo()>
      <Image
        source={
          Object {
            "testUri": "../../../src/images/chevron_right_corban.png",
          }
        }
        style={
          Object {
            "height": 24,
            "resizeMode": "contain",
            "width": 24,
          }
        }
      />
    </View>
    <Memo()
      key="place_holder_1"
    />
    <Memo()
      key="place_holder_2"
    />
    <Memo()
      key="place_holder_3"
    />
  </View>
</View>
`;
